SideBar.py

```diff
@@ -90,7 +90,73 @@ def copy(self, source, destination):
 			shutil.copy2(source, destination)
 
 		if self.view:
-			self.view.erase_status('ZZZ')
+			self.view.erase_status('SideBarTools:Copy')
 
 	def description(self):
 		return 'Duplicate File…'
+
+
+def temporary_status_message(view, message, key='SideBarTools', duration=5000):
+	view.set_status(key, message)
+
+	def erase_status():
+		view.erase_status(key)
+
+	sublime.set_timeout_async(erase_status, duration)
+
+
+class SideBarMoveCommand(SideBarCommand):
+
+	def run(self, paths):
+		self.view = self.window.active_view()
+		self.source = self.get_path(paths)
+
+		input_panel = self.window.show_input_panel(
+			'Move to:', self.source, self.on_done, None, None)
+
+		base, leaf = os.path.split(self.source)
+		name, ext = os.path.splitext(leaf)
+
+		input_panel.sel().clear()
+		input_panel.sel().add(sublime.Region(len(base) + 1, len(self.source) - len(ext)))
+
+	def on_done(self, destination):
+		threading.Thread(target=self.move, args=(self.source, destination)).start()
+
+	def move(self, source, destination):
+		print(source, destination)
+		if self.view:
+			self.view.set_status(
+				'SideBarTools:Move', 'Moving "{}" to "{}"'.format(source, destination))
+		else:
+			self.window.status_message(
+				'Moving "{}" to "{}"'.format(source, destination))
+
+		destination_dir = os.path.dirname(destination)
+		try:
+			os.makedirs(destination_dir)
+		except OSError:
+			print('Destination directory seems to exists...')
```

Comment:
Well there's an easy way of testing this:
- Create the test folder
- Change permissions so your current user can't write to it: assuming you use macOS: something along the lines of `chmod a-rwx,u+rwX,a+rX test-folder/` then `chown root:root test-folder/`; on POSIX platforms, those commands should make that folder only writeable by root, readable by everyone else.
- Try to move a file with the plugin, it should fail and you should get an exception or something in ST's console.

To alert the user about this error, we could use an actual message box, instead of status messages. This error is serious enough (should be rare enough) to force the user to dismiss the message and fix the permissions issue by other means.